MiniTwitch.Irc.Test/SpanSumTests.cs

```diff
@@ -0,0 +1,129 @@
+        { (int)Tags.ReplyThreadParentMsgId, "reply-thread-parent-msg-id" },
+        { (int)Tags.ReplyThreadParentUserLogin, "reply-thread-parent-user-login" },
+        { (int)Tags.SystemMsg, "system-msg" },
+        { (int)Tags.MsgParamColor, "msg-param-color" },
+        { (int)Tags.MsgParamMonths, "msg-param-months" },
+        { (int)Tags.MsgParamSubPlan, "msg-param-sub-plan" },
+        { (int)Tags.MsgParamSenderName, "msg-param-sender-name" },
+        { (int)Tags.MsgParamGiftMonths, "msg-param-gift-months" },
+        { (int)Tags.MsgParamViewerCount, "msg-param-viewerCount" },
+        { (int)Tags.MsgParamRecipientId, "msg-param-recipient-id" },
+        { (int)Tags.MsgParamSenderLogin, "msg-param-sender-login" },
+        { (int)Tags.MsgParamSenderCount, "msg-param-sender-count" },
+        { (int)Tags.MsgParamSubPlanName, "msg-param-sub-plan-name" },
+        { (int)Tags.MsgParamStreakMonths, "msg-param-streak-months" },
+        { (int)Tags.MsgParamMassGiftCount, "msg-param-mass-gift-count" },
+        { (int)Tags.MsgParamCumulativeMonths, "msg-param-cumulative-months" },
+        { (int)Tags.MsgParamRecipientUserName, "msg-param-recipient-user-name" },
+        { (int)Tags.MsgParamShouldShareStreak, "msg-param-should-share-streak" },
+        { (int)Tags.MsgParamRecipientDisplayName, "msg-param-recipient-display-name" },
+        { (int)Tags.MsgParamCharityName, "msg-param-charity-name" },
+        { (int)Tags.MsgParamDonationAmount, "msg-param-donation-amount" },
+        { (int)Tags.MsgParamExponent, "msg-param-exponent" },
+        { (int)Tags.MsgParamDonationCurrency, "msg-param-donation-currency" },
+        { (int)Tags.EmoteSets, "emote-sets" },
+        { (int)Tags.MessageId, "message-id" },
+        { (int)Tags.ThreadId, "thread-id" },
+    };
+
+    [Fact]
+    public void Test_Includes_All_Tags()
+    {
+        var tags = Enum.GetNames(typeof(Tags));
+        if (tags.Length != _values.Count)
+        {
+            var notIncluded = tags.Where(x => !_values.ContainsKey((int)Enum.Parse<Tags>(x)));
+            Assert.Fail($"Test does not account for all tags, expected {tags.Length}, got {_values.Count}." +
+                $"\nMissing tags: \n{string.Join("\n", notIncluded)}");
+        }
+
+        foreach (var tag in tags)
+        {
+            var enumValue = Enum.Parse<Tags>(tag);
+            try
+            {
+                Assert.True(_values.ContainsKey((int)enumValue));
+            }
+            catch (TrueException)
+            {
+                Assert.Fail($"Enum value {enumValue} not present.");
+            }
+        }
+    }
+
+    [Fact]
+    public void All_Tag_Values_Match_Sum()
+    {
+        foreach (var kvp in _values)
+        {
+            if (kvp.Key != MSum(kvp.Value))
+            {
+                Assert.Fail($"Tag \"{kvp.Value}\" does not match expected sum {kvp.Key}. Got {MSum(kvp.Value)} instead.");
+            }
+        }
+    }
+
+
+    private static int MSum(string s)
```

Comment:
This should really be a wrapped call to your implementation from `SpanExtensions`, you're only testing your test implementation of `MSum` here